content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,276 @@
+From the perspective of a cluster user, both autoscalers should provide a similar Node autoscaling
+experience. Both will provision new Nodes for unschedulable Pods, and both will consolidate the
+Nodes that are no longer optimally utilized.
+
+Different autoscalers may also provide features outside the Node autoscaling scope described on this
+page, and those additional features may differ between them.
+
+Consult the sections below, and the linked documentation for the individual autoscalers to decide
+which autoscaler fits your use-case better.
+
+#### Cluster Autoscaler
+
+Cluster Autoscaler adds or removes Nodes to pre-configured Node groups. Node groups generally map to
+some sort of provider-side VM group. A single instance of Cluster Autoscaler can simultaneously
+manage multiple Node groups. When provisioning, Cluster Autoscaler will add Nodes to the group that
+best fits the requests of pending Pods. When consolidating, Cluster Autoscaler always selects
+specific Nodes to remove as opposed to just resizing the underlying VM group.
+
+Additional context:
+
+* Documentation overview:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md
+* Supported cloud providers:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md#faqdocumentation
+* Extensive FAQ: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md
+* Contact: https://github.com/kubernetes/community/tree/master/sig-autoscaling#contact
+
+#### Karpenter
+
+Karpenter auto-provisions VMs based on [NodePool](https://karpenter.sh/docs/concepts/nodepools/)
+configuration provided by the user. Karpenter handles all aspects of node lifecycle, not just
+autoscaling. It works directly with VMs and doesn't rely on provider-side VM groups.
+
+Additional context:
+
+* Documentation: https://karpenter.sh/
+* Supported cloud providers:
+  https://github.com/kubernetes-sigs/karpenter?tab=readme-ov-file#supported-cloudproviders
+* Extensive FAQ: https://karpenter.sh/docs/faq/
+* Contact:
+  https://github.com/kubernetes-sigs/karpenter#community-discussion-contribution-and-support
+
+#### Implementation comparison
+
+Main differences between Cluster Autoscaler and Karpenter:
+
+* Cluster Autoscaler provides features related to just Node autoscaling. Karpenter has a wider
+  scope, and also provides features intended for managing Node lifecycle altogether (for example
+  grouping Nodes into pools, handling Node upgrades via disruption).
+* Cluster Autoscaler doesn't support auto-provisioning, the Node groups it can provision from have
+  to be pre-configured. Karpenter supports auto-provisioning, so the user only has to configure a
+  set of constraints for the provisioned Nodes instead of fully configuring homogenous groups.
+* Cluster Autoscaler provides integrations with numerous cloud providers, including smaller and less
+  popular providers. Karpenter currently only provides integrations with AWS and Azure.
```

Comment:
Maybe:
```suggestion
  popular providers. The Kubernetes project does not directly publish any autoscaler solution based
  on Karpenter, but third-party releases include integrations with AWS and Azure.
```

Reply:
Karpenter, within Kubernetes, is a library you can use to build a node autoscaler. Have I got that right?

Reply:
I can see what you're saying @sftim, but I think it's probably fair to just talk about Karpenter as it's been commonly implemented in cloud providers. Specifically in response to Tim's suggestion, here's what I'd propose:

```
* For Cluster Autoscaler, the Kubernetes project directly provides standalone Cloud Provider solutions. For Karpenter, the Kubernetes project publishes Karpenter as a core library of controllers with a common interface which is integrated and implemented by Cloud Providers to build a node autoscaler. Well known implementations include AWS and Azure.
```

Reply:
We need to make sure that the changes we merge don't conflict with requirements from our [content guide](https://kubernetes.io/docs/contribute/style/content-guide/).

**Also**, it's good to have content that is timeless. “Karpenter currently only provides integrations with AWS and Azure.” will go stale as soon as someone publishes a third integration.

---

I tweaked the text from https://github.com/kubernetes/website/pull/45802#discussion_r1588403053; @njtran WDYT?

```markdown
* For Cluster Autoscaler, the Kubernetes project directly provides standalone cloud provider
  integrations.
  For Karpenter, the Kubernetes project publishes Karpenter as a core library of controllers with
  a common interface that cloud providers can integrate and extend to build a node autoscaler.
  The implementations of Karpenter node autoscaling include AWS and Azure.
```

We don't limit ourselves to mentioning well known implementations; either we mention no third parties, or even a tiny cloud provider with four customers is welcome to mention their tool.

---

As a personal opinion, it'd be nice to see the CNCF landscape gain the ability to Kubernetes addons (perhaps after running a certification check to make sure they are compatible with stable Kubernetes?)

Reply:
Perhaps to add to this suggestion, we link out to the AWS and Azure implementations that we mention here?

Reply:
Simplified the suggestions from @sftim and @njtran, added links as @jonathan-innis suggested. WDYT about the current version?

Reply:
There's still a “currently”. Can we make this timeless and also avoid the risk that this goes stale due to an external change, such as someone using Karpenter to make an autoscaler?